godot-kotlin/godot-library/src/nativeMain/kotlin/godot/annotation/PropertyTypeHintAnnotation.kt

```diff
@@ -0,0 +1,246 @@
+package godot.annotation
+
+import godot.core.GodotResource
+import godot.registration.Range
+import kotlin.reflect.KClass
+
+/*
+All type checks will happen at the entry-generator at the moment.
+We should move that to the idea plugin at some point in time to provide IDE help and not just compilation errors.
+ */
+
+/**
+ * This property hint is optional!
+ *
+ * A Enum that is visible to the editor is always marked as PROPERTY_HINT_ENUM if not defined otherwise
+ * through another PropertyTypeHintAnnotation. It then will enumerate using strings.
+ *
+ * You only have to use this annotation if you want to enumerate enums with ints in the editor.
+ *
+ * Can only be used on Enum properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class EnumTypeHint(val enumerateAsInt: Boolean = false)
```

Comment:
What benefit do we get in `enumerateAsInt`? Or is it for the matter of completeness?

Reply:
Only for completeness